tests/torchtune/utils/test_pooling.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import pytest
+import torch
+from torchtune.utils.pooling import pool_sequence_logits
+
+
+class TestPooling:
+    def test_pool_sequence_logits_multi_batch(self):
+        """
+        Tests that the last non-padding token logits are pooled correctly for a multi-batch input.
+        """
+        padding_token_idx = 0
+        tokens = torch.tensor([[1, 3, 4, 9], [4, 5, 6, 0], [1, 0, 0, 0], [0, 0, 0, 0]])
+        logits = torch.tensor(
+            [
+                [[0.1, 1.3, 1.4], [0.5, 0.6, 0.7], [0.9, 1.1, 1.2], [1.3, 0.5, 1.6]],
+                [[0.2, 1.4, 1.5], [0.6, 0.7, 0.8], [1.0, 1.2, 1.3], [1.4, 1.6, 0.7]],
+                [[0.3, 1.5, 1.6], [0.1, 1.8, 0.2], [1.1, 1.3, 1.4], [0.5, 1.7, 0.1]],
+                [[0.4, 1.6, 1.7], [0.8, 0.9, 1.0], [1.2, 1.4, 1.5], [0.6, 1.8, 0.2]],
+            ]
+        )
+        expected_output = torch.tensor(
+            [
+                [1.3, 0.5, 1.6],
+                [1.0, 1.2, 1.3],
+                [0.3, 1.5, 1.6],
+                [0.4, 1.6, 1.7],
```

Comment:
I'm curious about this case: if all tokens are padding we return the first element? Any particular reason for this? (Hopefully something that won't happen very often in practice anyways)

Reply:
Yeah, this wasn't something I was super sure about. Agreed it'll be a weird edge case in practice. I think this is the implementation that makes the most sense to me - you always want the earliest token in the sequence pre-padding. If it's all padding, it feels consistent to just return the first token.

Reply:
Yeah it's true, the expected behavior is not really well-defined in this case. We definitely want to return something (don't want to get into the business of throwing out samples in the pooler), so I think your approach to return the first token makes sense.